src/hermit.rs

```diff
@@ -19,7 +19,7 @@ pub fn getrandom_inner(mut dest: &mut [MaybeUninit<u8>]) -> Result<(), Error> {
             dest = &mut dest[res as usize..];
         } else {
             let err = match res {
-                MIN_RET_CODE..=-1 => NonZeroU32::new(-res as u32).unwrap().into(),
```

Comment:
Note that if `res == -i32::MIN` then `-res` overflows.

Reply:
`MIN_RET_CODE` is `-i32::MAX`, so `-res` can not overflow. The current one results in a [slightly better codegen](https://rust.godbolt.org/z/aTxn7445v) and prevents accidental creation of errors in the custom range (e.g. if we got a corrupted return code for some reason), but your code is a bit easier to read and does not contain any `unwrap`s.

Reply:
This only relevant for Hermit, since it uses `isize` for return codes, while other targets use `i32`.